bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/values/ArrayValueImpl.java

```diff
@@ -346,8 +346,10 @@ public byte getByte(long index) {
         rangeCheckForGet(index, size);
         if (byteValues != null) {
             return byteValues[(int) index];
+        } else if (intValues != null) {
+            return ((Long) intValues[(int) index]).byteValue();
         }
-        return ((Integer) refValues[(int) index]).byteValue();
+        return ((Long) refValues[(int) index]).byteValue();
```

Comment:
Any particular reason to change this? Ideally they should fit in Integer right?

Reply:
A `byte` value in ballerina is represented by a Java `long`. So casting with `Integer` will give following class cast exception ```java.lang.ClassCastException: class java.lang.Long cannot be cast to class java.lang.Integer (java.lang.Long and java.lang.Integer are in module java.base of loader 'bootstrap')```